app/users/user.controller.js

```diff
@@ -0,0 +1,30 @@
+const HttpCodes = require('../httpcodes')
+const UserService = require('./user.service')
+
+const UserController = {
+    createUser,
+    getUserCount
+}
+
+function createUser(request, response) {
+    let userName = request.body.userName
+    let result = UserService.createUser(userName)
+    switch (result) {
+        case UserService.UserResult.Success:
+            response.send('User created')
+            return
+        case UserService.UserResult.UserExists:
+            response.status(HttpCodes.CONFLICT)
+            response.send(`User ${userName} already exists`)
+            return
+        default:
+            throw `Unknown UserResult ${result}`
```

Comment:
You need to be careful with throwing errors, as in this case this would crash your app. You'll need to define some catching for errors somewhere, in order to handle them gracefully (in this case I guess returning a 500 error response)

Reply:
I was a little unsure what to do here. I am used to kotlin enums. The compiler forces you to consider all enums but it is not the case with js. My thinking was that if a future developer(me) was to add another UserResult enum then they would get an early warning with an error being thrown in the development cycle and/or test suite.

Actually what happens here is that a 500 is returned. I guess express is handling the error?